evaluation/evaluator.go

```diff
@@ -0,0 +1,87 @@
+package evaluation
+
+import "fmt"
+
+type GoodnessFunction[Input, Output any] func(input Input, output Output) float64
+
+type Options[Input, Output any] struct {
+	GoodnessFunction GoodnessFunction[Input, Output]
+	Repetitions      int
+}
+
+type Tester[Input, Output any] interface {
+	Test(test Input) (Output, error)
+}
+
+type Evaluator[Input, Output any] struct {
+	options *Options[Input, Output]
+	tester  Tester[Input, Output]
+}
+
+func NewEvaluator[Input, Output any](tester Tester[Input, Output], options *Options[Input, Output]) *Evaluator[Input, Output] {
+	return &Evaluator[Input, Output]{
+		options: options,
+		tester:  tester,
+	}
+}
+
+func (e *Evaluator[Input, Output]) Evaluate(testPack []Input) ([]float64, error) {
+	channels := make([]chan []float64, e.options.Repetitions)
```

Comment:
Rename this to something a little more informative, like `repetitionScoreChannels`